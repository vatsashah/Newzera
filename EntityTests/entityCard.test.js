// Importing required library
import React from 'react'
import { shallow, mount } from 'enzyme'

// Importing component to be tested
import EntityCard from '../../../Components/Entity/entityCard'

describe('Entity Card should render without any error', () => {
  it('Entity Card should render correctly', () => {
    const component = shallow(<EntityCard />)
    expect(component).toMatchSnapshot()
  })
  it('Entity is not being followed', () => {
    const wrapper = mount(<EntityCard />).props()
    expect(wrapper.entityIsFollowing).toEqual(false)
  })
  it('Entity is being followed', () => {
    const wrapper = mount(<EntityCard entityIsFollowing />).props()
    expect(wrapper.entityIsFollowing).toEqual(true)
  })
})
