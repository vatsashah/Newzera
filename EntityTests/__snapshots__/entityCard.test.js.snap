// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entity Card should render without any error Entity Card should render correctly 1`] = `
<div
  className="entitycard"
  style={
    Object {
      "left": "0vw",
      "top": "0vh",
    }
  }
>
  <div>
    <img
      alt="Entity"
      className="entityimg"
      src="Trump.png"
    />
  </div>
  <div>
    <p
      className="entityname"
    >
      Donald Trump
    </p>
  </div>
  <div>
    <img
      alt="Trending Icon"
      className="entitytrending"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "icon-trending.svg",
        }
      }
    />
  </div>
  <div>
    <p
      className="entitydescription"
    >
      President of the USA
    </p>
  </div>
  <div>
    <p
      className="numFollowing"
    >
      54K Following
    </p>
  </div>
  <div
    className="PointerWhenOnHover share"
  >
    <ActionBubble
      activityType="Share"
    />
  </div>
  <div
    className="PointerWhenOnHover react"
  >
    <ActionBubble
      activityType="React"
    />
  </div>
  <div
    className="PointerWhenOnHover isfollowing"
  >
    <RectangleButton
      bordered={true}
      disabled={false}
      filled={false}
      fontSize={20}
      height={50}
      radius={38.5}
      title="Follow"
      width={204}
    />
  </div>
  <div>
    <p
      className="DetailTitle"
    >
      Details
    </p>
  </div>
  <div>
    <p
      className="entitydetail"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam.
    </p>
  </div>
</div>
`;
